python/paddle/distributed/fleet/meta_parallel/pipeline_parallel.py

```diff
@@ -2351,3 +2351,260 @@ def forward_backward_pipeline(
 
         self.timer_printer()
         return train_loss
+
+
+class PipelineParallelWithInterleaveFthenBMemoryEfficient(
+    PipelineParallelWithInterleaveFthenB
+):
+    def __init__(self, layers, hcg, strategy):
+        super().__init__(layers=layers, hcg=hcg, strategy=strategy)
+        logger.info("Using PipelineParallelWithInterleaveFthenBMemoryEfficient")
+
+    def forward_backward_pipeline(
+        self,
+        data,
+        scaler,
+        forward_only=False,
+        compute_loss=True,
+        return_micro_batch_loss=False,
+    ):
```

Comment:
If you need to support forward_only, you can simply let it go to the normal branch.

Reply:
> If you need to support forward_only, you can simply let it go to the normal branch.

done